### PR TITLE
feat: offline-mode

### PR DESCRIPTION
**Pull request by novakzaballa**
Implement offline mode for NodeJs SDK
Further documentation in the issue ticket https://github.com/Flagsmith/flagsmith/issues/2024

### CODE REVIEW

File: tests/sdk/flagsmith.test.ts [hunk 1]
```diff
@@ -275,18 +276,119 @@ test('getIdentitySegments throws error if identifier is empty string', () => {
...
+    await flagsmith.getEnvironmentFlags();
+    await flagsmith.getIdentityFlags('identity', {});
+
+    // Then
+    expect(mock_offline_handler.getEnvironment).toHaveBeenCalledTimes(1);
+    expect(flagsmith.getEnvironmentFlags).toHaveBeenCalled();
+    expect(flagsmith.getIdentityFlags).toHaveBeenCalled();
+  
+    const environmentFlags:Flags = await flagsmith.getEnvironmentFlags();
+    const identityFlags:Flags = await flagsmith.getIdentityFlags('identity', {});
```
Comment (matthewelwell) [inline tests/sdk/flagsmith.test.ts:349]: Is there a reason we don't just assign these values in the calls made a few lines above?

**Review by matthewelwell — COMMENTED** [review]
I've added one minor comment but this looks good in general.

**Comment by matthewelwell** [comment]
@novakzaballa we may also need to consider [this](https://github.com/Flagsmith/flagsmith-python-client/issues/66) in our implementation here too.

File: tests/sdk/flagsmith.test.ts (hunk 1 above)
Comment (novakzaballa) [reply tests/sdk/flagsmith.test.ts]: Oops, I really don't know why. Fixed

**Comment by novakzaballa** [comment]
> @novakzaballa we may also need to consider [this](https://github.com/Flagsmith/flagsmith-python-client/issues/66) in our implementation here too.

As per the condition at the beginning of this block, I would say that that's not the case in NodeJS.
https://github.com/Flagsmith/flagsmith-nodejs-client/blob/b1fc193f9f094fea42259befd4ff83561d371697/sdk/index.ts#L142-L177

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**